app/controllers/users_controller.rb

```diff
@@ -54,11 +60,32 @@ def set_user
 
     # Only allow a list of trusted parameters through.
     def user_params
-      params.require(:user).permit([:display_name, :full_name, :family_name, :orcid, :email_messages_enabled])
+      @user_params ||= params.require(:user).permit([:display_name, :full_name, :family_name, :orcid, :email_messages_enabled, { collections_with_messaging: {} }])
     end
 
     def can_edit?
       return true if current_user.id == @user.id
       current_user.super_admin?
     end
+
+    def parameter_enables_messaging?(form_value)
+      form_value.to_s == COLLECTION_MESSAGING_ENABLED
+    end
+
+    def update_collections_with_messaging
```

Comment:
Is there a way to break this method up? possibly move the contents of the loop into a method
```
      def configure_collection_notification(collection_id, param)
          selected_collection = Collection.find_by(id: collection_id)

          if parameter_enables_messaging?(param)
            @user.enable_messages_from(collection: selected_collection)
          else
            @user.disable_messages_from(collection: selected_collection)
          end
      end
```